### PR TITLE
Fix antialloca

### PR DESCRIPTION
**Pull request by tgymnich**

### CODE REVIEW

File: enzyme/Enzyme/GradientUtils.cpp [hunk 1]
```diff
@@ -3841,32 +3841,46 @@ Value *GradientUtils::invertPointerM(Value *const oval, IRBuilder<> &BuilderM,
...
-    auto dst_arg =
-        bb.CreateBitCast(antialloca, Type::getInt8PtrTy(oval->getContext()));
-    auto val_arg = ConstantInt::get(Type::getInt8Ty(oval->getContext()), 0);
-    auto len_arg =
-        ConstantInt::get(Type::getInt64Ty(oval->getContext()),
-                         M->getDataLayout().getTypeAllocSizeInBits(
-                             oval->getType()->getPointerElementType()) /
-                             8);
-    auto volatile_arg = ConstantInt::getFalse(oval->getContext());
+
```
Comment (wsmoses) [inline enzyme/Enzyme/GradientUtils.cpp:3858]: Can you fuse these?

Comment (tgymnich) [reply enzyme/Enzyme/GradientUtils.cpp]: done

**Review by wsmoses — APPROVED** [review]
Please add a test first, otherwise LGTM

**Comment by wsmoses** [comment]
@tgymnich please add a test

**Pull request merged**